testing/endpoint.go

```diff
@@ -573,11 +573,11 @@ func (endpoint *Endpoint) QueryChannelUpgradeProof() ([]byte, []byte, clienttype
 
 	// query proof for the channel on the counterparty
 	channelKey := host.ChannelKey(counterpartyPortID, counterpartyChannelID)
-	proofChannel, height := endpoint.Counterparty.Chain.QueryProof(channelKey)
+	proofChannel, height := endpoint.Counterparty.QueryProof(channelKey)
```

Comment:
The `Endpoint.QueryProof` will use the latest counterparty client state height. The `Endpoint.Chain.QueryProof` will use the latest chain height.

So if the chainA latest height did not equal the latest height of the client on chainB, this would fail. We can definitely improve function name naming here to make this more obvious

Reply:
re-naming would definitely be helpful. Preemptively changed in #3770 too.

Reply:
opened #3790